### ISSUE TITLE
training with triplet loss only (without softmax or center loss)

### ISSUE THREAD

**Issue by markncx** [JDAI-CV/fast-reid#489]
have you tried triplet loss only (without softmax or center loss). For my dataset (a little bit dirty), both softmax and softmax_triplet work, however, triplet loss cannot work. I find it hard to set the hyperparameters, and it takes a long time to try different sampling strategies. can anyone give some suggestions?

i prefer triplet because of the large number of ids(> 10 million, and is still growing). GPU memory is limited.

**Comment by L1aoXingyu** [comment]
Actually, training with triplet loss only needs more time to converge. You can refer to face recognition which has large number of ids (>100 million) and they still prefer softmax loss rather than triplet loss or pairwise contrastive loss.

**Comment by markncx** [comment]
> Actually, training with triplet loss only needs more time to converge. You can refer to face recognition which has large number of ids (>100 million) and they still prefer softmax loss rather than triplet loss or pairwise contrastive loss.

Thanks for your constructive comments. I'm using softmax-based loss for my task now. But the number of samples in each identity is not large (< 20), triplet loss maybe a better one in this case.

**Comment by L1aoXingyu** [comment]
According to our experiments, pairwise cosface is much better than triplet loss.

**Comment by markncx** [comment]
> According to our experiments, pairwise cosface is much better than triplet loss if the number of images in each id is low.

Thanks for your comprehensive comment. I'll try diffent loss functions for my task (image retrieval).

**Comment by markncx** [comment]
> According to our experiments, pairwise cosface is much better than triplet loss if the number of images in each id is low.

another question. does pairwise cosface outperform the conventional cosface?

**Comment by L1aoXingyu** [comment]
It depends on your data distribution. No Silver Bullet.
